lib/api.js

```diff
@@ -0,0 +1,124 @@
+var FS = require('fs')
+var OS = require('os')
+var Path = require('path')
+
+var glob = require('glob')
+var mkdirp = require('mkdirp')
+var rimraf = require('rimraf')
+var toId = require('to-id')
+
+var HOME_DIR = OS.homedir()
+var BASE_PATH = Path.join(HOME_DIR, 'hoodie')
+
+module.exports = {
+  add: function (properties) {
+    return createPackageJson(properties.name)
+
+    .then(function () {
+      return {
+        id: toId(properties.name),
+        name: toId(properties.name)
+      }
+    })
+  },
+  findAll: function () {
+    return assureHoodieFolder()
+
+    .then(function () {
+      return new Promise(function (resolve, reject) {
+        glob(BASE_PATH + '/*/package.json', function (error, packageJsonPaths) {
```

Comment:
I wasn't able to find 'packageJsonPaths' in node documentation. Is it customized parameter? Is it 'array'(I guessed with '.map' function as below)? I wonder which array will be passed to this parameter? ;)